tests/attacks/test_knockoff_nets.py

```diff
@@ -0,0 +1,434 @@
+
+        victim_preds = np.argmax(victim_ptc.predict(x=self.x_train[:100]), axis=1)
+        thieved_preds = np.argmax(thieved_ptc.predict(x=self.x_train[:100]), axis=1)
+        acc = np.sum(victim_preds == thieved_preds) / len(victim_preds)
+
+        self.assertGreater(acc, 0.3)
+
+        # Create adaptive attack
+        attack = KnockoffNets(classifier=victim_ptc, batch_size_fit=BATCH_SIZE, batch_size_query=BATCH_SIZE,
+                              nb_epochs=NB_EPOCHS, nb_stolen=NB_STOLEN, sampling_strategy='adaptive', reward='all')
+        thieved_ptc = attack.extract(x=self.x_train, y=self.y_train, thieved_classifier=thieved_ptc)
+
+        victim_preds = np.argmax(victim_ptc.predict(x=self.x_train[:100]), axis=1)
+        thieved_preds = np.argmax(thieved_ptc.predict(x=self.x_train[:100]), axis=1)
+        acc = np.sum(victim_preds == thieved_preds) / len(victim_preds)
+
+        self.assertGreater(acc, 0.4)
+        self.x_train = np.swapaxes(self.x_train, 1, 3)
+
+
+class TestKnockoffNetsVectors(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        (x_train, y_train), (_, _), _, _ = load_dataset('iris')
+
+        cls.x_train = x_train
+        cls.y_train = y_train
+
+    def setUp(self):
+        master_seed(1234)
+
+    def test_iris_tf(self):
+        """
+        First test for TF.
+        :return:
+        """
+        # Get the TF classifier
+        victim_tfc, sess = get_iris_classifier_tf()
+
+        # Define input and output placeholders
+        input_ph = tf.placeholder(tf.float32, shape=[None, 4])
+        output_ph = tf.placeholder(tf.int32, shape=[None, 3])
+
+        # Define the tensorflow graph
+        dense1 = tf.layers.dense(input_ph, 10)
+        dense2 = tf.layers.dense(dense1, 10)
+        logits = tf.layers.dense(dense2, 3)
+
+        # Train operator
+        loss = tf.reduce_mean(tf.losses.softmax_cross_entropy(logits=logits, onehot_labels=output_ph))
+        optimizer = tf.train.AdamOptimizer(learning_rate=0.001)
+        train = optimizer.minimize(loss)
+
+        # Tensorflow session and initialization
+        sess.run(tf.global_variables_initializer())
+
+        # Train the classifier
+        thieved_tfc = TensorFlowClassifier(clip_values=(0, 1), input_ph=input_ph, output=logits, labels_ph=output_ph,
+                                           train=train, loss=loss, learning=None, sess=sess, channel_index=1)
+
+        # Create random attack
+        attack = KnockoffNets(classifier=victim_tfc, batch_size_fit=BATCH_SIZE, batch_size_query=BATCH_SIZE,
+                              nb_epochs=NB_EPOCHS, nb_stolen=NB_STOLEN, sampling_strategy='random')
+        thieved_tfc = attack.extract(x=self.x_train, thieved_classifier=thieved_tfc)
+
+        victim_preds = np.argmax(victim_tfc.predict(x=self.x_train[:100]), axis=1)
+        thieved_preds = np.argmax(thieved_tfc.predict(x=self.x_train[:100]), axis=1)
```

Comment:
Would it be possible to test for the expected values of `victim_preds` and `thieved_preds`?

Reply:
For the victim_preds, it's no sense to test its expected value since it is just from the get_classifier_xx function and we made no change on it.
For the thieved_preds, it is impossible to test its expected value since we have many random factors from selecting data to the training of the thieved classifier.